File: 2536-increment-submatrices-by-one/2536-increment-submatrices-by-one.py
class Solution:
    def rangeAddQueries(self, n: int, queries: List[List[int]]) -> List[List[int]]:
        ans = [[0] * n for _ in range(n)]
        for [r1, c1, r2, c2] in queries:
            for i in range (r1, r2 + 1):
                ans[i][c1] += 1
                if c2 + 1 < n:
                    ans[i][c2 + 1] -= 1
        for i in range (0, n):
            for j in range (1, n):
                ans[i][j] += ans[i][j-1]
        return ans

File: 2536-increment-submatrices-by-one/2536-increment-submatrices-by-one.cpp
class Solution {
public:
    vector<vector<int>> rangeAddQueries(int n, vector<vector<int>>& queries) {
        vector<vector<int>> ans(n, vector<int>(n, 0));
        for (auto& cur : queries) {
            for (int i = cur[0]; i <= cur[2]; i++) {
                ans[i][cur[1]]++;
                if (cur[3] + 1 < n) {
                    ans[i][cur[3] + 1]--;
                }
            }
        }
        for (int i = 0; i < n; i++) {
            for (int j = 1; j < n; j++) {
                ans[i][j] += ans[i][j - 1];
            }
        }
        return ans;
    }
};